core/src/main/scala/spire/math/Algebraic.scala

```diff
@@ -66,86 +167,1383 @@ extends ScalaNumber with ScalaNumericConversions
+      }
+    } else {
+      s"Algebraic(~$approx)"
+    }
+  }
+
+  /**
+   * Returns the nearest, valid `Int` value to this Algebraic, without going
+   * further away from 0 (eg. truncation).
+   *
+   * If this `Algebraic` represented 1.2, then this would return 1. If this
+   * represented -3.3, then this would return -3. If this value is greater than
+   * `Int.MaxValue`, then `Int.MaxValue` is returned. If this value is less
+   * than `Int.MinValue`, then `Int.MinValue` is returned.
+   */
+  def intValue: Int = {
+    val n = toBigInt
+    if (n < MinIntValue) Int.MinValue
+    else if (n > MaxIntValue) Int.MaxValue
+    else n.intValue
+  }
+
+  /**
+   * Returns the nearest, valid `Long` value to this Algebraic, without going
+   * further away from 0 (eg. truncation).
+   *
+   * If this `Algebraic` represented 1.2, then this would return 1. If this
+   * represented -3.3, then this would return -3. If this value is greater than
+   * `Long.MaxValue`, then `Long.MaxValue` is returned. If this value is less
+   * than `Long.MinValue`, then `Long.MinValue` is returned.
+   */
+  def longValue: Long = {
+    val n = toBigInt
+    if (n < MinLongValue) Long.MinValue
+    else if (n > MaxLongValue) Long.MaxValue
+    else n.longValue
+  }
+
+  /**
+   * Returns a `Float` that approximates this value. If the exponent is too
+   * large to fit in a float, the `Float.PositiveInfinity` or
+   * `Float.NegativeInfinity` is returned.
+   */
+  def floatValue: Float = toBigDecimal(MathContext.DECIMAL32).toFloat
+
+  /**
+   * Returns a `Double` that approximates this value. If the exponent is too
+   * large to fit in a double, the `Double.PositiveInfinity` or
+   * `Double.NegativeInfinity` is returned.
+   */
+  def doubleValue: Double = toBigDecimal(MathContext.DECIMAL64).toDouble
+
+  /**
+   * Returns the nearest, valid `BigInt` value to this Algebraic, without going
+   * further away from 0 (eg. truncation).
+   *
+   * If this `Algebraic` represented 1.2, then this would return 1. If this
+   * represented -3.3, then this would return -3.
+   */
+  def toBigInt: BigInt =
+    toBigDecimal(0, RoundingMode.DOWN).toBigInt
+
+  /**
+   * Absolute approximation to `scale` decimal places with the given rounding
+   * mode. Rounding is always exact.
+   */
+  def toBigDecimal(scale: Int, roundingMode: RoundingMode): BigDecimal =
+    BigDecimal(roundExact(this, expr.toBigDecimal(scale + 2), scale, roundingMode))
+
+  /**
+   * Relative approximation to the precision specified in `mc` with the given
+   * rounding mode. Rounding is always exact. The sign is always correct; the
+   * sign of the returned `BigDecimal` matches the sign of the exact value this
+   * `Algebraic` represents.
+   *
+   * @param mc the precision and rounding mode of the final result
+   * @return an approximation to the value of this algebraic number
+   */
+  def toBigDecimal(mc: MathContext): BigDecimal = {
```

Comment:
This might be a case where we can use `Real` and `Algebraic` to test each other?